django_scylla/cql/compiler.py

```diff
@@ -1,14 +1,27 @@
 from time import time
+from random import SystemRandom
+from os import getpid
 
 from django.core.exceptions import EmptyResultSet
 from django.db import NotSupportedError
-from django.db.models import AutoField
+from django.db.models import AutoField, BinaryField, DecimalField, DurationField
 from django.db.models.sql import compiler
 
+from datetime import timedelta
+from cassandra.util import Duration
 
-def unique_rowid():
-    # TODO: guarantee that this is globally unique
-    return int(time() * 1e6)
+
+def unique_rowid(counter={'value': 0}):
+    """Attempt to return a reasonably random integer value with no collisions.
+
+    In the future, this behavior will likely be discouraged in favor of using UUIDs.
+    """
+    counter['value'] = (counter['value'] + 1) % 1024
+    timestamp = int(time() * 1e6)
+    pid = getpid()
+    sys_random = SystemRandom().randint(0, 65535)
+
+    return timestamp + counter['value'] + pid + sys_random
```

Comment:
Ideally this approach means that given the same code running simultaneously: each worker will return a different pid, each thread of that worker will return a different counter, and each call will return a different `sys_random` value.

The counter in particular is worth noting because it uses a little known "pitfall" of Python, which is that default values specified as dicts are persistent between runs.

e.g.

```python
>>> def foo(counter={'value':0}):
...   counter['value'] += 1
...   return counter['value']
...
>>> foo()
1
>>> foo()
2
>>> foo()
3
>>> foo()
4
```